component/discovery/discovery.go

```diff
@@ -2,20 +2,64 @@ package discovery
 
 import (
 	"context"
+	"fmt"
 	"sort"
 	"sync"
 	"time"
 
 	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/pkg/cluster"
 	"github.com/prometheus/prometheus/discovery"
 	"github.com/prometheus/prometheus/discovery/targetgroup"
 	"github.com/prometheus/prometheus/model/labels"
+	"github.com/rfratto/ckit/shard"
 )
 
+// Clustering determines whether to use clustering.
+type Clustering struct {
+	NodeUpdates bool `river:"node_updates,attr"`
```

Comment:
I think the River attribute name here is a little confusing. `enable_clustering`?